lib/api/src/modules/stories.ts

```diff
@@ -223,94 +223,99 @@ const initStoriesApi = ({
       kind.match(/\.|\|/)
     );
 
-    const storiesHashOutOfOrder = Object.values(input).reduce((acc, item) => {
-      const { kind, parameters } = item;
-      // FIXME: figure out why parameters is missing when used with react-native-server
-      const {
-        hierarchyRootSeparator: rootSeparator = undefined,
-        hierarchySeparator: groupSeparator = undefined,
-        showRoots = undefined,
-      } = { ...provider.getConfig(), ...((parameters && parameters.options) || {}) };
-
-      const usingShowRoots = typeof showRoots !== 'undefined';
-
-      // Kind splitting behaviour as per https://github.com/storybookjs/storybook/issues/8793
-      let root = '';
-      let groups: string[];
-      // 1. If the user has passed separators, use the old behaviour but warn them
-      if (typeof rootSeparator !== 'undefined' || typeof groupSeparator !== 'undefined') {
-        warnRemovingHierarchySeparators();
-        if (usingShowRoots) warnUsingHierarchySeparatorsAndShowRoots();
-        ({ root, groups } = parseKind(kind, {
-          rootSeparator: rootSeparator || '|',
-          groupSeparator: groupSeparator || /\/|\./,
-        }));
-
-        // 2. If the user hasn't passed separators, but is using | or . in kinds, use the old behaviour but warn
-      } else if (anyKindMatchesOldHierarchySeparators && !usingShowRoots) {
-        warnChangingDefaultHierarchySeparators();
-        ({ root, groups } = parseKind(kind, { rootSeparator: '|', groupSeparator: /\/|\./ }));
-
-        // 3. If the user passes showRoots, or doesn't match above, do a simpler splitting.
-      } else {
-        const parts: string[] = kind.split('/');
-        if (showRoots && parts.length > 1) {
-          [root, ...groups] = parts;
+    const { storySort = () => 0 } = provider.getConfig();
+
+    const storiesHashOutOfOrder = Object.entries(input)
+      .sort(storySort)
```

Comment:
Am I right in saying that if you set the `storySort` options parameter it sorts the story on the preview side (and so they'll be sorted e.g. in `store.extract()`), but if you set the addon option it sorts them only in the UI here?

That seem... super confusing, and likely to cause problems (if for Chromatic + other integrators if nothing else).

I would vote you just **CANNOT** set `storySort` in `manager.js`. It is a property of the store, not a UI option. With that in mind it probably shouldn't live in `parameters.options`. 🤷‍♂

Reply:
I'm considering putting it in `main.js` and making it a hard requirement it's a pure function.

That way I can inject the function into both environments...

Reply:
I'm not necessarily against putting it in `main.js` but why do we need it in both environments?

Reply:
I'm looking into this today.